### PR TITLE
Remove SourceToCloud feature from 2023-05-02-preview

### PR DESCRIPTION
**Pull request by pauld-msft**
# ARM (Control Plane) API Specification Update Pull Request 

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [ ] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [x] Other, please clarify:
    - _removing a feature from a new API version that has not been released yet_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I have reviewed the general guidance on the spec PR review process: https://aka.ms/specprreview.
- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data-plane related specifications.
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy).
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can request review from an ARM API Review board.

### ARM API changes review

- If you want for the ARM team to review this PR, you must add the `ARMReview` label. 
- The automation may automatically add the `ARMReview` label, if appropriate.  
  If this happens, proceed according to guidance given in GitHub comments also added by the automation.

### Breaking change review

If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), 
follow the process outlined in the [High-level Breaking Change Process doc](https://aka.ms/breakingchangesprocess#high-level-breaking-change-process).
      
## Getting help

- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For additional help, see https://aka.ms/azsdk/support/spectools.

### CODE REVIEW

**Review by daniv-msft — APPROVED** [review]

**Comment by pauld-msft** [comment]
@rkmanda not sure why the pipelines are not running, could you please kick them off? thanks

**Comment by pauld-msft** [comment]
Was able to get the pipelines to run by closing and reopening. @rkmanda could you please help review and merge this PR？

**Comment by pauld-msft** [comment]
For context, removed ARM review labels that I added manually, as we are removing a feature from an API version that has not been released. So this does not require an ARM reviewer

**Comment by pauld-msft** [comment]
Hi @rkmanda, was hoping to get this merged in soon if possible. Please let me know if there is someone else I should reach out to in order to get it merged. Thanks.

**Comment by konrad-jamrozik** [comment]
@pauld-msft I would recommend adding `WaitForARMFeedback` to get on the PR review queue which is [quite long](https://github.com/Azure/azure-rest-api-specs/pulls?page=1&q=is%3Aopen+is%3Apr+label%3AWaitForARMFeedback+-label%3AIDCDevDiv+draft%3Afalse). 

Details on why here:
[Roopesh Manda: Expect long delays for ARM swagger reviews](https://teams.microsoft.com/l/message/19:0351f5f9404446e4b4fd4eaf2c27448d@thread.skype/1688755962143?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1688755962143&teamName=Azure%20SDK&channelName=API%20Spec%20Review&createdTime=1688755962143)
posted in Azure SDK / API Spec Review on Friday, July 7, 2023 11:52 AM

File: specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/Builds.json [hunk 1]
```diff
@@ -1,486 +0,0 @@
-{
```
Comment (visingla-ms) [inline specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/Builds.json:1]: Why are these resource types being removed?

File: specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/Builders.json [hunk 2]
```diff
@@ -1,501 +0,0 @@
-{
```
Comment (visingla-ms) [inline specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/Builders.json:1]: Why are these resource types being removed?

File: specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/Builds.json (hunk 1 above)
Comment (daniv-msft) [reply specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/Builds.json:1]: Thanks @visingla-ms for your comments. These resource/resource types correspond to an API version that has not yet been released, and that we don't want to release anymore in the 2023-05-02-preview version.
As such, we're just removing these in this PR. In the future, we'll open a new PR once we know which version to target on our side. Again, this API version was never released with these changes in the first place.

File: specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/Builders.json (hunk 2 above)
Comment (daniv-msft) [reply specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/Builders.json:1]: Thanks for your comment. Please see my other message.

File: specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/Builds.json (hunk 1 above)
Comment (daniv-msft) [reply specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/Builds.json:1]: @konrad-jamrozik Tagging you as asked separately. These resource troupes are OK to remove (cf. above), and as such there are no more comments for us to address.

**Review by rkmanda — APPROVED** [review]

**Pull request merged**